{{cookiecutter.repo_name}}/.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,167 @@
+
+    - name: Build information
+      run: |
+        uname -a
+        df -h
+        ulimit -a
+
+{% if cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
+    - name: Set up Python {{ '${{ matrix.python-version }}' }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: {{ '${{ matrix.python-version }}' }}
+
+    - name: Install testing dependencies
+      # need non-login bash shell for Mac outside conda
+      # otherwise it sources system Python
+      shell: bash
+      run: |
+        python -m pip install -U pytest pytest-xdist pytest-cov codecov
+{% else %}
+    # More info on options: https://github.com/conda-incubator/setup-miniconda
+    - name: Install conda dependencies
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: {{ '${{ matrix.python-version }}' }}
+        environment-file: devtools/conda-envs/test_env.yaml
+        add-pip-as-python-dependency: true
+        architecture: x64
+{% if cookiecutter.dependency_source == 'Prefer conda-forge over the default anaconda channel with pip fallback' %}
+        mamba-version: "*"
+        channels: conda-forge, defaults
+{% elif cookiecutter.dependency_source == 'Prefer default anaconda channel with pip fallback' %}
+        channels: defaults
+{% endif %}
+        activate-environment: {{cookiecutter.repo_name}}-test
+        auto-update-conda: true
+        auto-activate-base: false
+        show-channel-urls: true
+{% endif %}
+
+{% if cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
+    - name: Install package
+      shell: bash
+      run: |
+        python --version
+        python -m pip install .
+{% else %}
+    - name: Install package
+      run: |
+        python -m pip install . --no-deps
+{% endif %}
+
+    - name: Python information
+{% if cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
+      shell: bash
+{% endif %}
+      run: |
+        which python
+        which pip
+        pip list
+{% if cookiecutter.dependency_source != 'Dependencies from pip only (no conda)' %}
+        conda info
+        conda list
+{% endif %}
+
+    - name: Run tests
+{% if cookiecutter.dependency_source == 'Dependencies from pip only (no conda)' %}
+      shell: bash
+{% endif %}
+      run: |
+        pytest -n 2 -v --cov={{ cookiecutter.repo_name }} --cov-report=xml --color=yes {{ cookiecutter.repo_name }}/tests/
+
+    - name: codecov
+      uses: codecov/codecov-action@v3
+      with:
+        file: coverage.xml
+        name: codecov-{{ '${{ matrix.os }}' }}-py{{ '${{ matrix.python-version }}' }}
+        verbose: True
+
+
+  pylint_check:
+    if: "github.repository == '{{ cookiecutter.github_username }}/{{ cookiecutter.repo_name }}'"
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.9
```

Comment:
to deal with later, is there a way for us to set a dynamic "python latest - 1" variable? that way we could just feed this whenever we have to declare a single python version anywhere (and not have to remember to update the cookiecutter).

Reply:
I've turned this into LAST_MINOR_PYTHON_RELEASE for now.